src/main/java/com/epam/msu/dao/impl/CertificateDaoImpl.java

```diff
@@ -0,0 +1,90 @@
+package com.epam.msu.dao.impl;
+
+import com.epam.msu.dao.CertificateDao;
+import com.epam.msu.dao.SqlRequest;
+import com.epam.msu.dao.mapper.CertificateMapper;
+import com.epam.msu.entity.Certificate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class CertificateDaoImpl implements CertificateDao {
+
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public CertificateDaoImpl(JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public List<Certificate> getAllCertificates() {
+        return jdbcTemplate.query(SqlRequest.getAllCertificates, new CertificateMapper());
+    }
+
+    @Override
+    public List<Certificate> getCertificatesByTagId(int tagId) {
+        return jdbcTemplate.query(SqlRequest.getCertificateByTagId,
+                new Object[]{tagId}, new BeanPropertyRowMapper<>(Certificate.class));
+    }
+
+    @Override
+    public Certificate getCertificateById(int id) {
```

Comment:
It is better not to return null(You can return optional to prevent null checks and possible nullPointers in the future). And also please fix not only this case, but look through the code